common/static/js/lang-edx.js

```diff
@@ -0,0 +1,65 @@
+var edx = edx || {},
+    Language = (function($) {
+        'use strict';
+        var preference_api_url = $('#preference-api-url').val(),
+            settings_language_selector = $('#settings-language-value'),
+            self = null;
+        return {
+
+            init: function() {
+                self = this;
+                this.listenForLanguagePreferenceChange();
+            },
+
+            /**
+             * Listener on changing language from selector.
+             * Send an ajax request to save user language preferences.
+             */
+            listenForLanguagePreferenceChange: function() {
+                settings_language_selector.change(function(event) {
+                    event.preventDefault();
+
+                    // ajax request to save user language preferences.
+                    self.sendAjaxRequest(preference_api_url, this.value, function () {
```

Comment:
Our best practice is to not use callbacks, but to use promises instead. ``$.ajax`` returns a promise so it is easy to then chain success and failure handlers to it.